consoles/serial_screen.pm

```diff
@@ -118,19 +104,13 @@ sub normalise_pattern ($pattern, $no_regex) {
         my $hr = shift @$pattern;
         if (@$pattern > 0) {
             my $re = qr/($hr)/;
-            for my $r (@$pattern) {
-                $re .= qr/|($r)/;
-            }
+            $re .= qr/|($_)/ for @$pattern;
```

Comment:
hm, looks like all this can be simplified (and corrected) by:
```
my $re = join "|", map { "($_)" } @$pattern;
return qr{$re};
```
This way also in the case of one element it will get parens `()`

Reply:
I wonder why the various patterns should all get their own parens.
`(a|b|c)` should be the same as `(a)|(b)|(c)`, and then the code which uses the regex can just use `$1` instead of `@@LAST_MATCH_START`

Reply:
The above does not work for me. Fails with

```
t/10-terminal.t .. 1/?   
#   Failed test '[Child 22536] direct: use array of regexs'
#   at t/10-terminal.t line 366.
#                   '
# 
# 
# Welcome to SUSE Linux Enterprise Server 12 SP2 RC3 (x86_64) - Kernel 4.4.21-65-default (hvc0).
# 
# linux-5rw7'
#     doesn't match '(?^u:(?^u:.*\
# \
# Welcome\ to\ SUSE\ Linux\ Enterprise\ Server\ 12\ SP2\ RC3\ \(x86_64\)\ \-\ Kernel\ 4\.4\.21\-65\-default\ \(hvc0\)\.\
# \
# linux\-5rw7\ login\:\ ))'
# Looks like you failed 1 test of 31.
```

Reply:
Then the test should be adjusted. The regex is needlessly verbose. But I guess that's for another PR.

Reply:
Yeah. Somehow I wasn't able to mark the whole block for my comment.